routing/matcher.go

```diff
@@ -358,8 +376,10 @@ func (m *matcher) match(r *http.Request) (*Route, map[string]string) {
 		path = path[:len(path)-1]
 	}
 
+	lrm := &leafRequestMatcher{r, path}
+
 	// first match fixed and wildcard paths
-	leaves, params := matchPathTree(m.paths, path)
+	leaves, params := matchPathTree(m.paths, path, lrm)
 	l := matchLeaves(leaves, r, path)
```

Comment:
Yes, it is possible, changes in pathmux will be needed (Matcher.Match, node.search, maybe other palces).

Also I see another one option. Just save latest matched leave in leafRequestMatcher (kind of cache) and take it in matcher.match.
Can we rely on such logic? I just fear that it will be lovely place for future bugs.
What do you think?

If you have same feelings, than I will commit changes later to pathmux and skipper.